xpra/codecs/nvenc/encoder.pyx

```diff
@@ -2340,11 +2606,34 @@ cdef class Encoder:
         cdef unsigned int stride = image.get_rowstride()
         log("do_compress_image(%s) kernel=%s, GPU buffer=%#x, stride=%i, input pitch=%i, output pitch=%i",
             image, self.kernel_name, int(gpu_buffer or 0), stride, self.inputPitch, self.outputPitch)
+
+        self.input_width = roundup(w, 32)
+        self.input_height = roundup(h, 32)
+        self.encoder_width = roundup(w, 32)
+        self.encoder_height = roundup(h, 32)
+
+        dims_changed = False
+        if USE_SINGLETON_ENCODER and self.instance_initialized():
```

Comment:
I think it would be worth having support for `dims_changed` even without the singleton encoder.